examples/AdvancedOperations/AddAppCampaign.php

```diff
@@ -0,0 +1,443 @@
+        //   1- you cannot override bid settings at the ad group level.
+        //   2- you cannot add ad group criteria
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars cruises ' . uniqid()]),
+            'status' => AdGroupStatus::ENABLED,
+            'campaign' => new StringValue(['value' => $campaignResourceName])
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Submits the ad group operation to add the ad group.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $response = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        $createdAdGroupResourceName = $response->getResults()[0]->getResourceName();
+        printf(
+            "Ad Group created '%s'%s",
+            $createdAdGroupResourceName,
+            PHP_EOL
+        );
+
+        // Creates an ad group ad.
+        $adGroupAd = new AdGroupAd([
+            'status' => AdGroupAdStatus::ENABLED,
+            'ad_group' => new StringValue(['value' => $createdAdGroupResourceName]),
+            'ad' => new Ad([
+                // ad_data is a 'oneof' message so setting app_ad
+                // is mutually exclusive with ad data fields such as
+                // text_ad, gmail_ad, etc.
+                'app_ad' => new AppAdInfo([
+                    'headlines' => [
+                        self::createAdTextAsset('A cool puzzle game'),
+                        self::createAdTextAsset('Remove connected blocks')
+                    ],
+                    'descriptions' => [
+                        self::createAdTextAsset('3 difficulty levels'),
+                        self::createAdTextAsset('4 colorful fun skins')
+                    ]
+                    // Optional: You can set up to 20 image assets for your campaign.
+                    // 'images' => [INSERT_AD_IMAGE_ASSET_ID(s)_HERE]
+                ])
+            ])
+        ]);
+
+        // Creates an ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Submits the ad group ad operation to add the ad group ad.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        $response = $adGroupAdServiceClient->mutateAdGroupAds($customerId, [$adGroupAdOperation]);
+
+        $createdAdGroupAdResourceName = $response->getResults()[0]->getResourceName();
+        printf(
+            "Ad Group App Ad created '%s'%s",
+            $createdAdGroupAdResourceName,
+            PHP_EOL
+        );
+    }
+
+    /**
+     * Creates an ad text asset for a given text.
+     *
+     * @param string $text the text to create the ad asset for
+     * @return string the created ad text asset
+     */
+    private static function createAdTextAsset(
```

Comment:
This is a small function, you can probably get rid of it and inline the code instead.

Reply:
I agree. @fiboknacky this sounds like a major change to me, I am happy to do it with PHP but not sure it is the best process to follow: I thin the golden needs to be updated first?

Reply:
For this one, I think it's a small detail so you can just go ahead and change only in this library.
It'll not be trivial if the method is really used or planned to be used in a guide.